microsoft/testsuites/xdp/functional.py

```diff
@@ -166,6 +166,47 @@ def verify_xdp_action_aborted(
             False,
         )
 
+    @TestCaseMetadata(
+        description="""
+        It validates XDP with different MTU
+
+        1. Check current image supports XDP or not.
+        2. change MTU to 1500, 2000, 3506 to test XDP.
+        """,
+        priority=3,
+        requirement=simple_requirement(min_count=2),
+    )
+    def verify_xdp_with_different_mtu(self, environment: Environment) -> None:
+        xdp_node = environment.nodes[0]
+        remote_node = environment.nodes[1]
+        xdpdump = self._get_xdpdump(xdp_node)
+        remote_address = self._get_ping_address(environment)
+        tested_mtu: List[int] = [1500, 2000, 3506]
+
+        xdp_node_nic_name = xdp_node.nics.default_nic
+        remote_nic_name = remote_node.nics.default_nic
+
+        xdp_node_ip = xdp_node.tools[Ip]
+        remote_ip = remote_node.tools[Ip]
+
+        original_xdp_node_mtu = xdp_node_ip.get_mtu(xdp_node_nic_name)
+        original_remote_mtu = remote_ip.get_mtu(remote_nic_name)
+
+        try:
+            for mtu in tested_mtu:
+                xdp_node_ip.set_mtu(xdp_node_nic_name, mtu)
+                remote_ip.set_mtu(remote_nic_name, mtu)
+
+                # tested mtu equals (ping mtu - IP headr (20) - ICMP header (8))
+                xdpdump.test(
+                    xdp_node_nic_name,
+                    remote_address=remote_address,
+                    ping_package_size=mtu - 28,
+                )
+        finally:
+            xdp_node_ip.set_mtu(xdp_node_nic_name, original_xdp_node_mtu)
```

Comment:
Should we mark the node to be recycled if we are unable to reset MTU back to its original value?

Reply:
The dirty env is not implemented yet. This line is safe here, it's in `finally` block, will be run in any case. And if the set_mtu will fail, the mtu won't be changed in previous logic too.